### PR TITLE
Web service supports both period and bracket notations

### PR DESCRIPTION
**Pull request by Hebcak**
Nested lists can be now written both with brackets and periods.
Ex. both address[0].city and address.0.city are possible

Part of: #91

### CODE REVIEW

**Comment by paveljandejsek** [comment]
Generated [reports](http://95.168.205.146:8080/reports/158/index.xml)

File: jbehave-support-core/src/main/java/org/jbehavesupport/core/ws/WebServiceHandler.java [hunk 1]
```diff
@@ -53,10 +55,12 @@
  * }
  * </pre>
  */
+
 public abstract class WebServiceHandler {
 
     private static final String CONTEXT_SEPARATOR = ".";
     private static final String REQUEST_POSTFIX = "Request";
+    public static final String BRACKET_REGEX = "(.*)\\[(\\d+)\\](\\.)(.*)";
```
Comment (paveljandejsek) [inline jbehave-support-core/src/main/java/org/jbehavesupport/core/ws/WebServiceHandler.java]: Nitpicking: do we need the selection groups if we dont use them? Might improve readability a bit

File: jbehave-support-core/src/main/java/org/jbehavesupport/core/ws/WebServiceHandler.java [hunk 2]
```diff
@@ -91,14 +95,29 @@ private void initTemplateInternal() {
...
+        ExamplesTable checkedData = checkCollectionNotation(data);
+        overrideRequestData(request, checkedData);
     }
 
     public final void overrideRequestData(String request, ExamplesTable data) {
         endpointRegistry.validateRequest(request);
         TestContextUtil.putDataIntoContext(testContext, data, request);
     }
 
+    private ExamplesTable checkCollectionNotation(ExamplesTable data) {
```
Comment (paveljandejsek) [inline jbehave-support-core/src/main/java/org/jbehavesupport/core/ws/WebServiceHandler.java]: convertCollectionNotation or something like that since its not just checking?

File: jbehave-support-core/src/main/java/org/jbehavesupport/core/ws/WebServiceHandler.java (hunk 1 above)
Comment (Hebcak) [reply jbehave-support-core/src/main/java/org/jbehavesupport/core/ws/WebServiceHandler.java]: From the code perspective they are unnecessary, however I used them to make sense of the regex, because it is a cluster of escaping special characters

**Comment by paveljandejsek** [comment]
Generated [reports](http://95.168.205.146:8080/reports/158/index.xml)

**Comment by paveljandejsek** [comment]
Generated [reports](http://95.168.205.146:8080/reports/158/index.xml)

**Comment by AmyKosinova** [comment]
Please add groovy test with mixed notation. This should verify that elements are not duplicated

File: jbehave-support-core/src/main/java/org/jbehavesupport/core/ws/WebServiceHandler.java [hunk 3]
```diff
@@ -91,14 +95,29 @@ private void initTemplateInternal() {
...
         endpointRegistry.validateRequest(request);
         TestContextUtil.putDataIntoContext(testContext, data, request);
     }
 
+    private ExamplesTable convertCollectionNotation(ExamplesTable data) {
+        List<Map<String, String>> newMapList = data.getRows().stream()
+            .map(map -> {
+                String name = map.get(ExampleTableConstraints.NAME);
+                if (name.matches(BRACKET_REGEX)) {
+                    String newName = name.replace("[", ".").replace("]", "");
```
Comment (paveljandejsek) [inline jbehave-support-core/src/main/java/org/jbehavesupport/core/ws/WebServiceHandler.java]: Will it work for nested indexes?

**Comment by paveljandejsek** [comment]
Generated [reports](http://95.168.205.146:8080/reports/158/index.xml)

File: jbehave-support-core/src/test/groovy/org/jbehavesupport/core/TestConfig.groovy [hunk 4]
```diff
@@ -98,14 +99,14 @@ class TestConfig {
...
-        RollingLogResolver rollingLogResolver = new SimpleRollingLogResolver();
+        RollingLogResolver rollingLogResolver = new SimpleRollingLogResolver()
         return new SshTemplate(passwordSetting, env.getProperty("ssh.timestampFormat"), rollingLogResolver)
     }
 
     @Bean
     @Qualifier("TEST")
     WebServiceHandler requestFactoryTestHandler() {
-        return new WebServiceHandler() {
+        return new TestWebServiceHandler(env) {
```
Comment (paveljandejsek) [inline jbehave-support-core/src/test/groovy/org/jbehavesupport/core/TestConfig.groovy]: Just a question - why was this change needed?

Comment (Hebcak) [reply jbehave-support-core/src/test/groovy/org/jbehavesupport/core/TestConfig.groovy]: The `TestWebServiceHandler` is set up to acces the mock app without the need for further configuration
